cis194/week2/zehua/src/Exercises.hs

```diff
@@ -0,0 +1,23 @@
+module Exercises where
+
+import Log
+
+-- ex1
+createLogMessage :: MessageType -> String -> [String] -> LogMessage
+createLogMessage mt ts msg = LogMessage mt (read ts :: TimeStamp) (unwords msg)
+
+parseWords :: [String] -> LogMessage
+parseWords ("I":t:xs)   = createLogMessage Info t xs
+parseWords ("W":t:xs)   = createLogMessage Warning t xs
+parseWords a@("E":e:t:xs)
+    | errno >= 1 && errno <= 100 = createLogMessage (Error errno) t xs
+    | otherwise                  = Unknown (unwords a)
+  where
+    errno = read e :: Int
+parseWords s            = Unknown (unwords s)
+
+parseMessage :: String -> LogMessage
+parseMessage = parseWords . words
+
+parse :: String -> [LogMessage]
+parse = fmap parseMessage . lines
```

Comment:
@shineli1984 question: should we use the generic functor `fmap` or the equivalent instance bindings?

Reply:
fmap is preferred. This means if you can switch the function instance to something else, it’ll still work.